java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -638,8 +681,14 @@ public static Table readParquet(ParquetOptions opts, HostMemoryBuffer buffer,
     assert len > 0;
     assert len <= buffer.getLength() - offset;
     assert offset >= 0 && offset < buffer.length;
-    return new Table(gdfReadParquet(opts.getIncludeColumnNames(),
-        null, buffer.getAddress() + offset, len, opts.timeUnit().getNativeId()));
+    long amount = opts.getSizeGuess();
```

Comment:
just thinking out loud, could `getSizeGuess` be `getSizeGuessOrElse(len * COMPRESSION_RATIO_ESTIMATE)` for example?